sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala

```diff
@@ -123,7 +123,11 @@ object JavaTypeInference {
         val beanInfo = Introspector.getBeanInfo(typeToken.getRawType)
         val properties = beanInfo.getPropertyDescriptors.filterNot(_.getName == "class")
         val fields = properties.map { property =>
-          val returnType = typeToken.method(property.getReadMethod).getReturnType
+          val readMethod = Option(property.getReadMethod).getOrElse {
+            throw new UnsupportedOperationException(
+              s"Cannot read the property ${property.getName} because it does not have the getter")
```

Comment:
nit: I think `the class` is better than `the property` because `property` is internal and it is some meaningless for users? In the bottom, an error message uses `class` https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/JavaTypeInference.scala#L433.